src/Raven.Server/ServerWide/Commands/Indexes/PutIndexCommand.cs

```diff
@@ -50,6 +58,31 @@ public override string UpdateDatabaseRecord(DatabaseRecord record, long etag)
             return null;
         }
 
+        private bool InitializeStaggeredDeployment(DatabaseRecord record)
+        {
+            // What happens here though? No members.. do we resend the command? How do we make sure this will run again?
+            if (record.Topology.Members.Count == 0)
+                return false;
+
+            record.StaggeredIndexes ??= new Dictionary<string, Dictionary<string, StaggeredIndexStatus>>();
+
+            if (record.StaggeredIndexes.TryAdd(Definition.Name, new Dictionary<string, StaggeredIndexStatus>()) == false)
+            {
+                //staggered index already being deployed, no need to add it to indexes again
+                return false;
+            }
+
+            // It's guaranteed that topology members are ordered the same on all nodes
```

Comment:
Not _quite_, mind. It is the same if they are in sync, they don't have to be.
And generally speaking, we simply need to use one, we don't actually care which.

Reply:
The order of the raft commands is guaranteed to be the same on all nodes.
So the order of the members will be the same on each state machine when executing this command.

I think we do want to use a node which isn't the primary to avoid changing the primary frequently if this index will consume a lot of resources.